src/Analysis/Engine/Impl/Values/Protocols.cs

```diff
@@ -71,7 +71,7 @@ abstract class Protocol : AnalysisValue, IHasRichDescription {
         public override IAnalysisSet GetDescriptor(PythonAnalyzer projectState, AnalysisValue instance, AnalysisValue context) => AnalysisSet.Empty;
         public override IAnalysisSet GetEnumeratorTypes(Node node, AnalysisUnit unit) => AnalysisSet.Empty;
         public override IAnalysisSet GetIndex(Node node, AnalysisUnit unit, IAnalysisSet index) => AnalysisSet.Empty;
-        public override IAnalysisSet GetInstanceType() => AnalysisSet.Empty;
+        public override IAnalysisSet GetInstanceType() => null;
```

Comment:
Looks dangerous. I think we have a lot of places that expect `IAnalysisSet` to be not null.

Reply:
This is to restore null since code that determines instance type of the protocol depends on it. We have two tests failing in master b/c of this.